internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -487,3 +497,27 @@ func (r *BundleReconciler) updateStatus(ctx context.Context, orig *fleet.Bundle,
 	metrics.BundleCollector.Collect(ctx, bundle)
 	return nil
 }
+
+// cleanupOrphanedBundleDeployments will delete all existing BundleDeployments which do not have a match in a provided list of UIDs
+func (r *BundleReconciler) cleanupOrphanedBundleDeployments(ctx context.Context, bundle *fleet.Bundle, uidsToKeep sets.Set[types.UID]) error {
+	list := &fleet.BundleDeploymentList{}
+	if err := r.List(ctx, list,
+		client.MatchingLabels{
+			fleet.BundleLabel:          bundle.GetName(),
+			fleet.BundleNamespaceLabel: bundle.GetNamespace(),
+		},
+	); err != nil {
+		return err
+	}
+
+	var errs []error
+	for _, bd := range list.Items {
+		if uidsToKeep.Has(bd.UID) {
+			continue
+		}
+		if err := r.Delete(ctx, &bd); client.IgnoreNotFound(err) != nil {
+			errs = append(errs, err)
+		}
```

Comment:
```suggestion
		if err := r.Delete(ctx, &bd); client.IgnoreNotFound(err) != nil {
			errs = append(errs, err)
		}
		
		if err := finalize.PurgeContent(ctx, r.Client, bd.Name, bd.Spec.DeploymentID); err != nil {
		        errs = append(errs, err)
		}
```

Reply:
I've added 39ac379 with that I think is the proper way to handle this (let the reconciliation from adding a `deletionTimestamp` to the `BundleDeployment` handle the Content management).
Please let me know if you feel strongly about this!

Reply:
Oh of course, that's much better! :trophy: 
The only amendment I would make to it is removing calls to `PurgeContent` from [here](https://github.com/rancher/fleet/blob/main/internal/cmd/controller/finalize/finalize.go#L103), and we'd be good to go.